agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/AiBackCompatSampler.java

```diff
@@ -7,18 +7,22 @@
 import io.opentelemetry.api.common.Attributes;
 import io.opentelemetry.api.trace.Span;
 import io.opentelemetry.context.Context;
-import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.data.LinkData;
 import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingDecision;
 import io.opentelemetry.sdk.trace.samplers.SamplingResult;
-import io.opentelemetry.sdk.trace.samplers.SamplingResult.Decision;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public final class TraceIdBasedSampler implements Sampler {
+// this sampler does two things:
+// * implements same trace id hashing algorithm so that traces are sampled the same across multiple nodes
+//   when some of those nodes are being monitored by other Application Insights SDKs (and 2.x Java SDK)
+// * adds sampling percentage to span attribute (TODO this is not being applied to child spans)
+public final class AiBackCompatSampler implements Sampler {
 
-    private static final Logger logger = LoggerFactory.getLogger(TraceIdBasedSampler.class);
+    private static final Logger logger = LoggerFactory.getLogger(AiBackCompatSampler.class);
```

Comment:
can we rename this too? AiLegacySampler?

Reply:
ya, renamed to `AiSampler` (it's not really legacy since there's no replacement in AI for it)